support/k8s/k8s-workload-registrar/mode-crd/controllers/pod_controller.go

```diff
@@ -161,32 +184,85 @@ func (r *PodReconciler) updateorCreatePodEntry(ctx context.Context, pod *corev1.
 }
 
 // podSpiffeID returns the desired spiffe ID for the pod, or nil if it should be ignored
-func (r *PodReconciler) podSpiffeID(pod *corev1.Pod) string {
+func (r *PodReconciler) podSpiffeID(pod *corev1.Pod) (string, error) {
 	if r.c.PodLabel != "" {
 		// the controller has been configured with a pod label. if the pod
 		// has that label, use the value to construct the pod entry. otherwise
 		// ignore the pod altogether.
 		if labelValue, ok := pod.Labels[r.c.PodLabel]; ok {
-			return makeID(r.c.TrustDomain, "%s", labelValue)
+			return makeID(r.c.TrustDomain, "%s", labelValue), nil
 		}
-		return ""
+		return "", nil
 	}
 
 	if r.c.PodAnnotation != "" {
 		// the controller has been configured with a pod annotation. if the pod
 		// has that annotation, use the value to construct the pod entry. otherwise
 		// ignore the pod altogether.
 		if annotationValue, ok := pod.Annotations[r.c.PodAnnotation]; ok {
-			return makeID(r.c.TrustDomain, "%s", annotationValue)
+			return makeID(r.c.TrustDomain, "%s", annotationValue), nil
 		}
-		return ""
+		return "", nil
 	}
 
 	// the controller has not been configured with a pod label or a pod annotation.
+	if r.c.IdentityTemplate != "" {
+		// create an entry using provided identity template.
+		svid, err := r.getIdentityTemplate(pod)
+		if err != nil {
+			return "", err
+		}
+		return makeID(r.c.TrustDomain, svid), nil
+	}
+
+	// the controller has not been configured with a specific identity format.
 	// create an entry based on the service account.
-	return makeID(r.c.TrustDomain, "ns/%s/sa/%s", pod.Namespace, pod.Spec.ServiceAccountName)
+	return makeID(r.c.TrustDomain, "ns/%s/sa/%s", pod.Namespace, pod.Spec.ServiceAccountName), nil
 }
 
 func (r *PodReconciler) podParentID(nodeName string) string {
 	return makeID(r.c.TrustDomain, "k8s-workload-registrar/%s/node/%s", r.c.Cluster, nodeName)
 }
+
+func (r *PodReconciler) getIdentityTemplate(pod *corev1.Pod) (string, error) {
+	tpl := r.c.IdentityTemplate
+	tmpl, err := template.New("IdentityTemplate").Parse(tpl)
+	if err != nil {
+		r.c.Log.WithError(err).Errorf("error parsing the template %q", tpl)
+		return "", err
+	}
+
+	// Create the IdentityMaps struct, with maps, one for Pod and one for Context:
+	podInfo := map[string]string{
+		PodServiceAccountIDLabel: pod.Spec.ServiceAccountName,
+		NamespaceIDLabel:         pod.Namespace,
+		PodNameIDLabel:           pod.Name,
+		PodUIDLabel:              string(pod.UID),
+		PodHostnameLabel:         pod.Spec.Hostname,
+		PodNodeNameLabel:         pod.Spec.NodeName,
+	}
```

Comment:
Ive always seen templates use a standard struct that holds the desired values. Using the struct you can have different types, ie `types.UID`, rather than just all `string`. Why did you use map?